src/org/opensolaris/opengrok/configuration/Configuration.java

```diff
@@ -827,6 +838,25 @@ private static Configuration decodeObject(InputStream in) throws IOException {
         if (!(ret instanceof Configuration)) {
             throw new IOException("Not a valid config file");
         }
+        
+        for (Iterator<Group> it = ((Configuration)ret).getGroups().iterator(); it.hasNext();) {
```

Comment:
for (Group g:getgroups() ) ...

if you use Netbeans, please notice the left side yellow/orange exclamation marks - hints - usually it should make you aware

Also this check is overengineering(not sure how did you come to such overly complex cycle in cycle ;) ... KISS!) - if you need a deduplicated list, use appropriate structure and make sure your object properly does hashCode and equals implementation